scm-record fails build without default features

### Description of the bug

```
$ cargo build --no-default-features
   Compiling scm-record v0.1.0 (/usr/local/google/home/martinvonz/git-branchless/scm-record)
warning: unused import: `fs`
  --> scm-record/src/ui.rs:14:11
   |
14 | use std::{fs, io, mem, panic};
   |           ^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: unused imports: `DUMP_UI_STATE_FILENAME`, `ENV_VAR_DUMP_UI_STATE`
  --> scm-record/src/ui.rs:33:21
   |
33 | use crate::consts::{DUMP_UI_STATE_FILENAME, ENV_VAR_DEBUG_UI, ENV_VAR_DUMP_UI_STATE};
   |                     ^^^^^^^^^^^^^^^^^^^^^^                    ^^^^^^^^^^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared crate or module `serde_json`
  --> scm-record/src/types.rs:46:29
   |
46 |     SerializeJson(#[source] serde_json::Error),
   |                             ^^^^^^^^^^ use of undeclared crate or module `serde_json`

For more information about this error, try `rustc --explain E0433`.
warning: `scm-record` (lib) generated 2 warnings
error: could not compile `scm-record` (lib) due to previous error; 2 warnings emitted
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### Version of `rustc`

_No response_

### Automated bug report

_No response_

### Version of `git-branchless`

_No response_

### Version of `git`

_No response_

--- a/scm-record/src/types.rs
+++ b/scm-record/src/types.rs
@@ -42,6 +42,7 @@ pub enum RecordError {
     #[error("failed to read user input: {0}")]
     ReadInput(#[source] crossterm::ErrorKind),
 
+    #[cfg(feature = "serde")]
     #[error("failed to serialize JSON: {0}")]
     SerializeJson(#[source] serde_json::Error),
 

